proxy/src/control/destination/background.rs

```diff
@@ -102,7 +106,8 @@ pub(super) fn task(
     dns_resolver: dns::Resolver,
     namespaces: Namespaces,
     host_and_port: Option<HostAndPort>,
-    controller_tls: tls::ConditionalConnectionConfig<tls::ClientConfigWatch>
+    controller_tls: tls::ConditionalConnectionConfig<tls::ClientConfigWatch>,
+    sensors: &telemetry::Sensors,
```

Comment:
This gives more evidence to the idea that `ConnectionConfig` should just hold the sensors.